pallets/msa/src/lib.rs

```diff
@@ -720,6 +762,70 @@ impl<T: Config> Pallet<T> {
 		Ok(())
 	}
 
+	/// Revokes schema permissions
+	pub fn revoke_schema_permissions_for(
+		delegator: Delegator,
+		provider: Provider,
+		schema_ids: Vec<SchemaId>,
+	) -> DispatchResult {
+		DelegatorAndProviderToDelegation::<T>::try_mutate(
+			delegator,
+			provider,
+			|maybe_info| -> DispatchResult {
+				let mut info = maybe_info.take().ok_or(Error::<T>::DelegationNotFound)?;
+
+				Self::ensure_all_schema_ids_are_valid(&schema_ids)?;
+
+				let mut schema_permissions = info.schema_permissions;
+
+				let current_block = frame_system::Pallet::<T>::block_number();
+
+				for id in schema_ids {
+					schema_permissions.try_insert(id, Some(current_block.clone())).map_err(
+						|_| Error::<T>::ExceedsMaxSchemaGrantsPerDelegationOrSchemaNotGranted,
```

Comment:
looks like one cannot tell the difference between failing for exceeding the bound or a key not found.